operator/src/main/java/oracle/kubernetes/operator/logging/OncePerMessageLoggingFilter.java

```diff
@@ -0,0 +1,48 @@
+// Copyright 2019, Oracle Corporation and/or its affiliates.  All rights reserved.
+// Licensed under the Universal Permissive License v 1.0 as shown at
+// http://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.logging;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/** A LoggingFilter that logs each log message, which are typically message keys, at most once */
+public class OncePerMessageLoggingFilter implements LoggingFilter {
+
+  // allow all messages to be logged when filtering is off
+  volatile boolean filtering = false;
+
```

Comment:
filtering and the hashmap are protected by two different locks -- it's probably fine for this change since I beleive that instances of this class are only exercised single threaded - but still, this means that multi-threading class won't work in all situations.   Would be safer if filtering setter was protected by the same synchronize.